### PR TITLE
docs: updated contributing guidelines

### PR DESCRIPTION
**Pull request by jsam**
This PR updates a contribution guidelines and proposes new (more strict) way of specifying commit messages.

The benefits of being more strict with types and scopes are:
  1) possibility of auto generating releases end to end
  2) nicer way of grouping scopes in our change logs

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -7,16 +7,22 @@ little bit helps, and credit will always be given.
 Types of Contributions
 ----------------------
 
-Report Bugs
-~~~~~~~~~~~
+Report issues / contacting developers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: make sure the linter won't complain about this - looks too long to my eyes :)

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -7,16 +7,22 @@ little bit helps, and credit will always be given.
...
 
-Report bugs at https://github.com/SwissDataScienceCenter/renku-python/issues.
+Report bugs at our issue tracker_.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]:
```suggestion
Report bugs on our issue tracker_.
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -30,6 +36,13 @@ Implement Features
 Look through the GitHub issues for features. Anything tagged with "feature"
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: I don't think we actually tag anything with "feature"

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -30,6 +36,13 @@ Implement Features
...
 is open to whoever wants to implement it.
 
+
+Improvement requests
+~~~~~~~~~~~~~~~~~~~~
+
+If you see some of the things could be done in a better way, feel free to submit a issue with a improvement suggestion.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: wording suggestion:

If you see room for improvement, please open an issue with a suggestion.

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -30,6 +36,13 @@ Implement Features
...
+In suggestion you might include motivation for improvement and why new solution would be improvement.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: Wording suggestion:

Please motivate your suggestion by illustrating a problem it solves.

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -7,16 +7,22 @@ little bit helps, and credit will always be given.
...
-Report Bugs
-~~~~~~~~~~~
+Report issues / contacting developers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Report bugs at https://github.com/SwissDataScienceCenter/renku-python/issues.
+Report bugs at our issue tracker_.
 
-If you are reporting a bug, please include:
+If you want to submit a bug, improvement or feature suggestions feel free to open a corresponding issue on GitHub.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: please make sure to keep consistent line length

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -62,29 +75,29 @@ Ready to contribute? Here's how to set up `renku` for local development.
 
       $ git clone git@github.com:your_name_here/renku.git
 
-3. Install your local copy into a virtualenv. Assuming you have
-   virtualenvwrapper installed, this is how you set up your fork for local
-   development:
+3. Ensure you have your development environment set up. For this we encourage usage of `pipenv`
+(other virtual environment tools could be used as well) `pyenv`:
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: `pyenv` is kind of coming out of the blue for me here - aren't you talking about pipenv?

File: CONTRIBUTING.rst [hunk 8]
```diff
@@ -62,29 +75,29 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+      $ git checkout -b <issue_number>_<short_description>
 
    Now you can make your changes locally.
 
 5. When you're done making changes, check that your changes pass tests:
 
    .. code-block:: console
 
-      $ ./run-tests.sh
+      $ pipenv run tests
```
Comment (rokroskar) [inline CONTRIBUTING.rst]:
```suggestion
      $ pipenv run ./run-tests.sh
```

File: CONTRIBUTING.rst [hunk 9]
```diff
@@ -121,16 +134,49 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+-------------------------
+
+This project is using conventional commits style for generation of changelog upon each release. Therefore,
+it's important that our commit messages convey what they do correctly. Commit message should always follow this pattern:
+
+   $ %{type}(%{scope}): %{description}
+
+**Type's used for describing commit's which will end up in changelog are** :code:`fix:` & :code:`feat:`.
+
+Please note, that fix type here is only for user facing bug fixes and not fixes on tests or CI.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]:
```suggestion
Please note that the ``fix`` type here is only for user facing bug fixes and not fixes on tests or CI.
```

File: CONTRIBUTING.rst [hunk 10]
```diff
@@ -121,16 +134,49 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+
+Full list of types which are in use:
+  * :code:`feat:` - Used for new user facing features. This should be related to one of the predefined scopes. If a scope does not exist, new scope should be proposed for a adoption.
+  * :code:`fix:` - Used for fixing user facing bugs. This should be related to one of the predefined scopes.
+  * :code:`chore:` - Used for change which are not user facing. Scope should be a module name in which chore occurred.
+  * :code:`test:` - Used for fixing existing or adding new tests. Scope should relate to a module name which is being tested.
+  * :code:`docs:` - Used for adding more documentation. If documentation is not related to predefined user scopes, it can be omitted.
+  * :code:`refactor` - Used for changing the code structure. Scope being used here should be module name. If refactoring is across multiple modules, scope could be omitted or PR broken down into smaller chunks.
+
+Full list of user facing scope's which are in use:
```
Comment (rokroskar) [inline CONTRIBUTING.rst]:
```suggestion
Full list of user facing scopes which are in use:
```

File: CONTRIBUTING.rst [hunk 11]
```diff
@@ -121,16 +134,49 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+
+   $ %{type}(%{scope}): %{description}
+
+**Type's used for describing commit's which will end up in changelog are** :code:`fix:` & :code:`feat:`.
+
+Please note, that fix type here is only for user facing bug fixes and not fixes on tests or CI.
+For those, please use: :code:`ci:` or :code:`test:`
+
+Full list of types which are in use:
+  * :code:`feat:` - Used for new user facing features. This should be related to one of the predefined scopes. If a scope does not exist, new scope should be proposed for a adoption.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]:
```suggestion
  * :code:`feat:` - Used for new user facing features. This should be related to one of the predefined scopes. If a scope does not exist, a new scope may be proposed.
```

File: CONTRIBUTING.rst [hunk 12]
```diff
@@ -121,16 +134,49 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+  * :code:`fix:` - Used for fixing user facing bugs. This should be related to one of the predefined scopes.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]:
```suggestion
  * :code:`fix:` - Used for fixing user-facing bugs. This should be related to one of the predefined scopes.
```

File: CONTRIBUTING.rst [hunk 13]
```diff
@@ -121,16 +134,49 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+  * :code:`chore:` - Used for change which are not user facing. Scope should be a module name in which chore occurred.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]:
```suggestion
  * :code:`chore:` - Used for changes which are not user facing. The scope should be a module name in which chore occurred.
```

File: CONTRIBUTING.rst [hunk 14]
```diff
@@ -121,16 +134,49 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+  * :code:`test:` - Used for fixing existing or adding new tests. Scope should relate to a module name which is being tested.
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: should we use ``tests``?

```suggestion
  * :code:`test:` - Used for fixing existing or adding new tests. The scope should relate to a module name which is being tested.
```

File: CONTRIBUTING.rst [hunk 15]
```diff
@@ -138,25 +139,26 @@ Ready to contribute? Here's how to set up `renku` for local development.
 Commit message guidelines
 -------------------------
 
-This project is using conventional commits style for generation of changelog upon each release. Therefore,
-it's important that our commit messages convey what they do correctly. Commit message should always follow this pattern:
+This project is using conventional commits style for generation of changelog upon
```
Comment (rokroskar) [inline CONTRIBUTING.rst]: Maybe add a link to conventional commits?

Comment (jsam) [reply CONTRIBUTING.rst]: Added in #4fd255

File: CONTRIBUTING.rst (hunk 8 above)
Comment (rokroskar) [reply CONTRIBUTING.rst]: Oh I didn't realize this wasn't changed - it's currently incorrect

File: CONTRIBUTING.rst [hunk 16]
```diff
@@ -121,16 +136,52 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+**Type's used for describing commit's which will end up in changelog are** :code:`fix:` & :code:`feat:`.
+
+Please note that the :code:`fix` type here is only for user-facing bug fixes and not fixes on tests or CI.
+For those, please use: :code:`ci:` or :code:`test:`
+
+Full list of types which are in use:
+  * :code:`feat:` - Used for new user-facing features. This should be related to one of the predefined scopes. If a scope does not exist, a new scope may be proposed.
+  * :code:`fix:` - Used for fixing user-facing bugs. This should be related to one of the predefined scopes.
+  * :code:`chore:` - Used for changes which are not user-facing. The scope should be a module name in which chore occurred.
+  * :code:`tests:` - Used for fixing existing or adding new tests. The scope should relate to a module name which is being tested.
```
Comment (mohammad-sdsc) [inline CONTRIBUTING.rst]: Since we are testing a feature here the scope should come from a predefined scope too.

File: CONTRIBUTING.rst [hunk 17]
```diff
@@ -121,16 +136,52 @@ Ready to contribute? Here's how to set up `renku` for local development.
...
+  * :code:`docs:` - Used for adding more documentation. If documentation is not related to predefined user scopes, it can be omitted.
+  * :code:`refactor` - Used for changing the code structure. Scope being used here should be module name. If refactoring is across multiple modules, scope could be omitted or PR broken down into smaller chunks.
+
+Full list of user-facing scopes which are in use:
+  * :code:`graph` - Scope for describing knowledge graph which is being build with users usage of the system.
+  * :code:`workflow` - Scope for describing reproducibility flow.
+  * :code:`dataset` - Scope for describing datasets.
+  * :code:`core` - General scope for describing all core elements of Renku project.
+  * :code:`svc` - General scope for describing interaction or operation of Renku as a service.
```
Comment (mohammad-sdsc) [inline CONTRIBUTING.rst]: I prefer `service` instead of `svc`.

File: CONTRIBUTING.rst (hunk 16 above)
Comment (jsam) [reply CONTRIBUTING.rst]: Well, not always. Tests could be for a given utility functions which could be used across multiple scopes, that's why imho it makes more sense per module level. But I guess omitting the scope would make most sense in this case?

Comment (mohammad-sdsc) [reply CONTRIBUTING.rst]: Yes, that also makes sense to me.

File: CONTRIBUTING.rst (hunk 17 above)
Comment (rokroskar) [reply CONTRIBUTING.rst]: At the risk of things being more verbose, I second that

**Pull request merged**